File: app/models/panel/pwa.py
"""Admin panel DTOs: the installable-webapp (PWA) branding form."""

from __future__ import annotations

from pydantic import Field

from app.models.panel.common import ActionResponse, PanelRequest, PanelResponse


class PwaSettingsResponse(PanelResponse):
    app_name: str = ""
    short_name: str = ""
    description: str = ""
    has_custom_icon: bool = False
    icon_version: int = 0


class PwaSettingsSaveRequest(PanelRequest):
    app_name: str = Field("", max_length=45)
    short_name: str = Field("", max_length=12)
    description: str = Field("", max_length=300)


class PwaIconUploadResponse(ActionResponse):
    icon_version: int = 0

File: frontend/src/types/panel/pwa.ts
/** Mirrors app/models/panel/pwa.py */
import type { ActionResponse, PanelAuthRequest, PanelEnvelope } from "./common";

export interface PwaSettingsResponse extends PanelEnvelope {
  app_name: string;
  short_name: string;
  description: string;
  has_custom_icon: boolean;
  icon_version: number;
}

export interface PwaSettingsSaveRequest extends PanelAuthRequest {
  app_name: string;
  short_name: string;
  description: string;
}

export interface PwaIconUploadResponse extends ActionResponse {
  icon_version: number;
}
